src/bsoncxx/document/value.cpp

```diff
@@ -23,6 +23,9 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace document {
 
+value::value() noexcept : _data(NULL, [](std::uint8_t*) {}) {
```

Comment:
I'm for default constructable `document::value`'s whether or not the `Value` for the template is optional. We have similar default constructors for (e.g.) database, client, collection.